R/FilteredData.R

```diff
@@ -511,39 +504,46 @@ FilteredData <- R6::R6Class( # nolint
     #'        mae = list(dataset = miniACC)
     #'   )
     #' )
-    #' fs <- list(
-    #'   iris = list(
-    #'     Sepal.Length = list(selected = c(5.1, 6.4), keep_na = TRUE, keep_inf = FALSE),
-    #'     Species = list(selected = c("setosa", "versicolor"), keep_na = FALSE)
-    #'   ),
-    #'   mae = list(
-    #'     subjects = list(
-    #'       years_to_birth = list(selected = c(30, 50), keep_na = TRUE, keep_inf = FALSE),
-    #'       vital_status = list(selected = "1", keep_na = FALSE),
-    #'       gender = list(selected = "female", keep_na = TRUE)
-    #'     ),
-    #'     RPPAArray = list(
-    #'       subset = list(ARRAY_TYPE = list(selected = "", keep_na = TRUE))
-    #'     )
+    #' fs <-
+    #'   filter_settings(
+    #'     filter_var(dataname = "iris", varname = "Sepal.Length", selected = c(5.1, 6.4), keep_na = TRUE, keep_inf = FALSE),
+    #'     filter_var(dataname = "iris", varname = "Species", selected = c("setosa", "versicolor"), keep_na = FALSE),
+    #'     filter_var(dataname = "mae", varname = "years_to_birth", selected = c(30, 50), keep_na = TRUE, keep_inf = FALSE, target = "subjects"),
+    #'     filter_var(dataname = "mae", varname = "vital_status", selected = "1", keep_na = FALSE, target = "subjects"),
+    #'     filter_var(dataname = "mae", varname = "gender", selected = "female", keep_na = TRUE, target = "subjects"),
+    #'     filter_var(dataname = "mae", varname = "ARRAY_TYPE", selected = "", keep_na = TRUE, experiment = "RPPAArray", target = "subset")
     #'   )
-    #' )
     #' shiny::isolate(datasets$set_filter_state(state = fs))
     #' shiny::isolate(datasets$get_filter_state())
     #'
     set_filter_state = function(state) {
-      checkmate::assert_subset(names(state), self$datanames())
-      lapply(names(state), function(dataname) {
+      if (is.teal_slices(state)) {
+        checkmate::assert_class(state, "teal_slices")
+        lapply(self$datanames(), function(x) {
+          relevant_slices <- extract_by_feat(state, "dataname", x)
+          private$get_filtered_dataset(x)$set_filter_state(relevant_slices)
+        })
+      } else {
+        warning(paste(
+          "From FilteredData:",
+          "Specifying filters as lists is obsolete and will be deprecated in the next release.",
+          "Please see ?set_filter_state and ?filter_settings for details."
+        ),
+        call. = FALSE)
+        checkmate::assert_subset(names(state), self$datanames())
+        lapply(names(state), function(dataname) {
+          logger::log_trace(
+            "FilteredData$set_filter_state initializing, dataname: { paste(names(state), collapse = ' ') }"
+          )
+
+          fdataset <- private$get_filtered_dataset(dataname = dataname)
+          dataset_state <- state[[dataname]]
+          fdataset$set_filter_state(state = dataset_state)
```

Comment:
In case of passing old-nested-list down to FilterStates we should just convert them to `teal_slices` object. You can create utils function which convert `list->teal_slices`. Once we fully deprecate it will require less changes.

```suggestion
```

Reply:
That would mean the conversion only happens at the top level (`FilteredData`) and everything downstream works on `teal_slices`. Is this your intent?

Reply:
Yes. FilteredData is the only concern. Other classes are not intended to be used directly

Reply:
All right, will do :+1: